src/TestRunnerOrchestrator.ts

```diff
@@ -0,0 +1,146 @@
+        }
+      }
+      return new PromisePool(promiseProducer, testRunners.length).start();
+    });
+  }
+
+  private calculateTimeout(baseTimeout: number) {
+    return (this.options.timeoutFactor * baseTimeout) + this.options.timeoutMs;
+  }
+
+  private updateMutantStatus(mutant: Mutant, runResult: RunResult) {
+    switch (runResult.result) {
+      case TestResult.Timeout:
+        mutant.status = MutantStatus.TIMEDOUT;
+        break;
+      case TestResult.Complete:
+        if (runResult.failed > 0) {
+          mutant.status = MutantStatus.KILLED;
+        } else {
+          mutant.status = MutantStatus.SURVIVED;
+        }
+        break;
+    }
+    mutant.specsRan = runResult.specNames;
+  }
+
+  private runSingleTestsRecursive(testSelector: TestSelector, testRunner: IsolatedTestRunnerAdapter, runResults: RunResult[], currentTestIndex: number)
+    : Promise<RunResult[]> {
+    return new Promise<RunResult[]>(resolve => {
+      testSelector.select([currentTestIndex])
+        .then(() => testRunner.run({ timeout: 10000 }))
+        .then(runResult => {
+          if (runResult.result === TestResult.Complete && (runResult.succeeded > 0 || runResult.failed > 0)) {
+            runResults[currentTestIndex] = runResult;
+            resolve(this.runSingleTestsRecursive(testSelector, testRunner, runResults, currentTestIndex + 1));
+          } else {
+            testRunner.dispose();
+            resolve(runResults);
+          }
+        });
+    })
+  }
+
+  private createTestRunners(): Promise<TestRunnerMetadata[]> {
+
+    return new Promise<TestRunnerMetadata[]>((resolve, reject) => {
+      let cpuCount = os.cpus().length;
+      let testRunnerMetadatas: TestRunnerMetadata[] = [];
+      let allPromises: Promise<any>[] = [];
+      for (let i = 0; i < cpuCount; i++) {
+        allPromises.push(this.createTestRunner(i).then(testRunnerMetadata => testRunnerMetadatas.push(testRunnerMetadata)));
+      }
+      Promise.all(allPromises).then(() => resolve(testRunnerMetadatas));
+    });
+  }
+
+  private createTestRunner(portOffset: number): Promise<TestRunnerMetadata> {
+    return this.copyAllSourceFilesToTempFolder().then(sourceFileMap => {
+      let selector = TestSelectorFactory.instance().create(this.options.testFrameork, { options: this.options });
+      let tempSourceFiles: string[] = [];
+      for (let i in sourceFileMap) {
+        tempSourceFiles.push(sourceFileMap[i]);
+      }
+      return {
+        sourceFileMap,
+        runnerAdapter: IsolatedTestRunnerAdapterFactory.create(this.createTestRunSettings(tempSourceFiles, selector, this.options.port + portOffset, false)),
+        selector
+      };
+    });
+  }
+
+  private copyAllSourceFilesToTempFolder() {
+    return new Promise<FileMap>((resolve, reject) => {
+      let fileMap: FileMap = Object.create(null);
+      var tempFolder = StrykerTempFolder.createRandomFolder('test-runner-source-files');
```

Comment:
Also: I still have my doubts about the `StrykerTempFolder` being in the api folder and about the name. We should discuss this before V1